railties/lib/rails/source_annotation_extractor.rb

```diff
@@ -16,24 +22,32 @@ class SourceAnnotationExtractor
     # Wraps a regular expression that will be tested against each of the source
     # file's comments.
     class ParserExtractor < Struct.new(:pattern)
-      class Parser < Ripper
-        attr_reader :comments, :pattern
-
-        def initialize(source, pattern:)
-          super(source)
-          @pattern = pattern
-          @comments = []
+      if defined?(Prism)
+        def annotations(file)
+          Prism.parse_file(file).comments.filter_map do |comment|
+            Annotation.new(comment.location.start_line, $1, $2) if comment.location.slice =~ pattern
+          end
```

Comment:
It looks like the ripper parser is purposefully excluding files with syntax errors. If we want to match that behavior, you would need to check `success?`, like:

```suggestion
          result = Prism.parse_file(file)
          return [] unless result.success?
          
          result.comments.filter_map do |comment|
            Annotation.new(comment.location.start_line, $1, $2) if comment.location.slice =~ pattern
          end
```

Reply:
Thank you for checking, just added the change: https://github.com/rails/rails/pull/52287/commits/22df0d7ac6717cd17c727bb344b1db08b2336dbf

Reply:
Looks good!